### PR TITLE
add testing sdk docs

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

File: docs/developers/sdk/overview.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Overview
 
-**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers as well as custom in-house solutions into a SQL database.
-It was developed to allow easy creation for providers and reduce boilerplate code when developing them.
+**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers APIs as well as custom in-house solutions into a SQL database.
+It was developed to allow easy creation for providers and reduce boilerplate code when developing them, letting developers focus only on the **E**xtract while the SDK will take care of the rest.
```
Comment (roneli) [inline docs/developers/sdk/overview.md:4]:
```suggestion
It was developed to allow easy creation for providers and reduce boilerplate code when developing them, giving developers solely on the **E**xtract while the SDK takes care of the rest. 
```

File: docs/developers/sdk/overview.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 # Overview
 
-**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers as well as custom in-house solutions into a SQL database.
-It was developed to allow easy creation for providers and reduce boilerplate code when developing them.
+**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers APIs as well as custom in-house solutions into a SQL database.
+It was developed to allow easy creation for providers and reduce boilerplate code when developing them, letting developers focus only on the **E**xtract while the SDK will take care of the rest. 
 
-The core idea of the SDK is to allow developing a provider in a straightforward manner, while allowing the user free control in fetching the data to be inserted.
+The core idea of the SDK is to allow for developers developing a provider in a straightforward manner, while allowing the user free control in fetching the data to be inserted.
```
Comment (roneli) [inline docs/developers/sdk/overview.md:6]:
```suggestion
The core idea of the SDK is to allow developing a provider in a straightforward manner, while allowing the user free control in fetching the data to be inserted.
```

feels redundant

File: docs/developers/sdk/testing.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Testing
+
+Testing your provider is key in ensuring quality of your provider and developer experiance for future contributors that can easily test their changes against the tests.
```
Comment (roneli) [inline docs/developers/sdk/testing.md:3]:
```suggestion
Testing providers is key factor in ensuring the quality of your provider and developer experience, so future contributors can easily test their changes against.
```

File: docs/developers/sdk/testing.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+CloudQuery SDK provides utility test function to help test each resource and verify it's output.
```
Comment (roneli) [inline docs/developers/sdk/testing.md:5]:
```suggestion
CloudQuery SDK provides an utility test function to help you test each resource and verify it's output.
```

File: docs/developers/sdk/testing.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Requirements
+
+The testing framework provided by CloudQuery SDK is essentially an end-to-end (integration) tests. This means you will need a real test account and real resources in place. It is suggested to create those test resource with any IaC tool (Terraform, Pulumi, etc...).
```
Comment (roneli) [inline docs/developers/sdk/testing.md:9]:
```suggestion
The testing framework provided by CloudQuery SDK are essentially an end-to-end (integration) tests. This means you will need a real test account and real resources in place. It is suggested to create those test resources with any IaC tool (Terraform, Pulumi, etc...).
```
either `are ... e2e tests` or is ... e2e test`

File: docs/developers/sdk/testing.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+## How Resource Tests work
+
+There are two main goals to CloudQuery testing framework:
+
+- Check that the extract(fetch) functioanality is defined correctly and works without errors against real infrastructures.
+- Check the schema definition and all columns resolvers that are defined correctly and the data exists after a successful fetch.
+
+Here is the flow of each test:
```
Comment (roneli) [inline docs/developers/sdk/testing.md:18]: Maybe  subtitle it?

```suggestion
### Test flow execution:
```

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, added some comments, you are missing a mention to the IgnoreInTests flag

File: docs/developers/sdk/testing.md (hunk 4 above)
Comment (yevgenypats) [reply docs/developers/sdk/testing.md:5]: an utility ? a utility maybe ?

**Comment by yevgenypats** [comment]
@roneli I mentioned the IgnoreInTest flag for column and rows. I don't mention it for `TestResource` because it's just temporary. it will only be available in the table and column as otherwise we will have to duplicate structs and have long nested structs in `TestResource` for resources with nested children.

File: docs/developers/sdk/overview.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # Overview
 
-**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers as well as custom in-house solutions into a SQL database.
-It was developed to allow easy creation for providers and reduce boilerplate code when developing them.
+**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service providers APIs as well as custom in-house solutions into a SQL database.
```
Comment (disq) [inline docs/developers/sdk/overview.md:3]:
```suggestion
**The CloudQuery SDK** enables building CloudQuery providers which allows CloudQuery's users to **E**xtract/**T**ransform/**L**oad existing and popular service provider APIs as well as custom in-house solutions into a SQL database.
```

File: docs/developers/sdk/overview.md [hunk 8]
```diff
@@ -1,7 +1,7 @@
...
+It was developed to allow easy creation for providers and reduce boilerplate code when developing them, giving developers to solely focus on the **E**xtract while the SDK takes care of the rest.
```
Comment (disq) [inline docs/developers/sdk/overview.md:4]:
```suggestion
It was developed to allow easy creation for providers and reduce boilerplate code when developing them, enabling developers to solely focus on the **E**xtract while the SDK takes care of the rest. 
```

File: docs/developers/sdk/testing.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+# Testing
+
+Testing providers is key factor in ensuring the quality of your provider and developer experience, so future contributors can easily test their changes against.
```
Comment (disq) [inline docs/developers/sdk/testing.md:3]: against... ? Smells like an incomplete sentence, I think

File: docs/developers/sdk/testing.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+CloudQuery SDK provides a utility test function to help you test each resource and verify it's output.
+
+## Requirements
+
+The testing framework provided by CloudQuery SDK are essentially an end-to-end (integration) tests. This means you will need a real test account and real resources in place. It is suggested to create those test resources with any IaC tool (Terraform, Pulumi, etc...).
```
Comment (disq) [inline docs/developers/sdk/testing.md:9]:
```suggestion
The testing framework provided by CloudQuery SDK are essentially end-to-end (integration) tests. This means you will need a real test account and real resources in place. It is suggested to create those test resources with any IaC tool (Terraform, Pulumi, etc...).
```

File: docs/developers/sdk/testing.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+## How Resource Tests work
+
+There are two main goals to CloudQuery testing framework:
+
+- Check that the extract(fetch) functioanality is defined correctly and works without errors against real infrastructures.
+- Check the schema definition and all columns resolvers that are defined correctly and the data exists after a successful fetch.
```
Comment (disq) [inline docs/developers/sdk/testing.md:16]:
```suggestion
- Check the schema definition and all column resolvers are defined correctly, the data exists after a successful fetch.
```

File: docs/developers/sdk/testing.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+### Test flow execution:
+
+- The resource table and all it's children tables are delted from PostgreSQL to work with clean data.
```
Comment (disq) [inline docs/developers/sdk/testing.md:20]:
```suggestion
- The resource table and all it's children tables are deleted from PostgreSQL to work with a clean state.
```

File: docs/developers/sdk/testing.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+- CloudQuery SDK is fetching the resource, transforming and loading it into PostgreSQL.
```
Comment (disq) [inline docs/developers/sdk/testing.md:21]:
```suggestion
- CloudQuery SDK fetches the resource, transforms and loads it into PostgreSQL.
```

File: docs/developers/sdk/testing.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+- For each resource table (and it's children) CloudQuery is reading all the rows and checking that
```
Comment (disq) [inline docs/developers/sdk/testing.md:22]:
```suggestion
- For each resource table (and it's children) CloudQuery reads all the rows and checks that 
```

File: docs/developers/sdk/testing.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+- The resource table and all it's children tables are delted from PostgreSQL to work with clean data.
+- CloudQuery SDK is fetching the resource, transforming and loading it into PostgreSQL.
+- For each resource table (and it's children) CloudQuery is reading all the rows and checking that 
+    - Each table has at least one row available unless `IgnoreInTests` is defined in the `Table`.
+    - Each column has data which means the default or custom column resolver worked (unless skipped with `IgnoreInTests` on the `Column`).
+
+## ResourceTest and ResourceTestCase
+
+Each resource can be tested end-to-end with `ResourceTest` function. Here is an example
```
Comment (disq) [inline docs/developers/sdk/testing.md:28]:
```suggestion
Each resource can be tested end-to-end with `ResourceTest` function. Here is an example:
```

**Review by disq — CHANGES_REQUESTED** [review]

**Pull request merged**